src/Common/TagStoreInterface.php

```diff
@@ -4,15 +4,66 @@
 
 interface TagStoreInterface
 {
-    public function tagObject($objectId, array $tags);
+    /**
+     * Associate the given tags to cache item identified by $itemId.
+     *
+     * Example: associate a HTTP cache identifier (e.g. a content-digest in the
+     *          case of the Symfony HTTP cache) with a set of tags.
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function tagItemId($itemId, array $tags);
 
-    public function untagObject($objectId, array $tags);
+    /**
+     * Unassociate the given tags from the cache item identified by $itemId.
+     *
+     * NOTE: This is difficult for an so-called "cache based" implementation, is it necessary?
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function untagItemId($itemId, array $tags);
```

Comment:
In a file/cache based implementation this operation is not trivial, i.e. the $tag is the cache-key/filename, and the cache ids are serialized in the cache-entry/file - so searching is very inefficient without an index.

Is this (and other methods relying on searching for an item-id) necessary in the context of HTTP caching?